### PR TITLE
Switch from serve to pure nginx

### PR DESCRIPTION
**Pull request by nikklassen**
Using nginx allows us to set the cache-control headers for the service-worker specifically

### CODE REVIEW

**Comment by nikklassen** [comment]
Blocked on https://github.com/danReynolds/SummonerExpert/issues/84

File: docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1 @@
 version: '3'
```
Comment (danReynolds) [inline docker-compose.yml]: This file can probably be deleted

File: nginx.upstream.conf [hunk 2]
```diff
@@ -1,23 +1,14 @@
...
-
-  location @summonerexpert {
-    proxy_set_header Host $host;
-    proxy_set_header X-Forwarded-Proto $scheme;
+  root /app;
+  try_files $uri /index.html;
 
-    proxy_pass http://summonerexpert;
+  location ~* ^/(service-worker.js|index.html)$ {
+    add_header Cache-Control "max-age=0, no-cache, no-store, must-revalidate";
```
Comment (danReynolds) [inline nginx.upstream.conf]: Can we add a comment with a link to where it talks about needing this for service-workers?

File: Dockerfile [hunk 3]
```diff
@@ -4,5 +4,8 @@ FROM starefossen/ruby-node:2-8
 RUN mkdir /app
 WORKDIR /app
 
+# Add Ruby dependencies necessary for deployment
+RUN gem install sshkit rake
```
Comment (danReynolds) [inline Dockerfile]: You can remove the ruby stuff from dockerfile

**Pull request merged**